test/execute/index.js

```diff
@@ -0,0 +1,151 @@
+ * - "fail MESSAGE" will t.fail(MESSAGE).
+ * - "plan NUMBER_OF_TESTS" will t.plan(Number(NUMBER_OF_TESTS)).
+ * - "end" will t.end().
+ *
+ * A good strategy to follow is to SAY "plan NUMBER_OF_TESTS" first. Then
+ * "pass" and "fail" depending on expected scratch results in conditions, event
+ * scripts, or what is best for testing the target block or group of blocks.
+ * When its done you must SAY "end" so the test and tap know that the end has
+ * been reached.
+ */
+fs.readdirSync(__dirname)
+    .filter(uri => uri.endsWith('.sb2'))
+    .forEach(uri => {
+        test(uri, t => {
+            // Disable logging during this test.
+            log.suggest.deny('vm', 'error');
+            t.tearDown(() => log.suggest.clear());
+
+            const project = readFileToBuffer(path.resolve(__dirname, uri));
+
+            const vm = new VirtualMachine();
+            vm.attachStorage(makeTestStorage());
+
+            // Delay recording the early vm error with tap until plan has a
+            // chance to be set.
+            let vmError;
+            const testVmError = () => {
+                t.doesNotThrow(() => {
+                    if (vmError) {
+                        throw vmError;
+                    }
+                });
+            };
+
+            // Listen to SAY events and match the beginning of SAY text's to
+            // basic tap methods.
+            let didPlan;
+            let didEnd;
+            vm.runtime.on('SAY', (target, type, text) => {
+                if (text.toLowerCase().startsWith('pass')) {
+                    t.pass(text.substring(('pass').length).trim());
+                } else if (text.toLowerCase().startsWith('fail')) {
+                    t.fail(text.substring(('fail').length).trim());
+                } else if (text.toLowerCase().startsWith('plan')) {
+                    didPlan = true;
+                    t.plan(Number(text.substring(('plan').length).trim()) + 1);
+                    testVmError();
+                } else if (text.toLowerCase().startsWith('end')) {
+                    didEnd = true;
+                    if (!didPlan) {
+                        testVmError();
+                    }
+                    t.end();
+                } else {
+                    t.comment(text);
+                }
+            });
+
+            // Start the VM, load the project, and wait until all threads
+            // complete.
+            let loadPromise;
+            try {
+                vm.start();
+                vm.clear();
+                vm.setCompatibilityMode(false);
+                vm.setTurboMode(false);
+
+                // Stop the runtime interval once the test is complete so the
+                // test process may naturally exit.
+                t.tearDown(() => {
+                    clearInterval(vm.runtime._steppingInterval);
+                });
+
+                loadPromise = vm.loadProject(project)
+                    .then(() => vm.greenFlag())
```

Comment:
I think we could move this down to after the `try` block and attach it to the `return loadPromise.` because this is really `wholeTestDonePromise` now...

Reply:
In that case we could just get rid of the try block and t.doesNotThrow tests and use the tap's default error reporting I guess.